src/options.c

```diff
@@ -442,10 +443,9 @@ void optionsParseThumbnail(char* optarg)
     }
 }
 
-void optionsParseFileName(char* fileName)
+void optionsParseFileName(const char* optarg)
 {
-    opt.outputFile = fileName;
-
+    opt.outputFile = strdup(optarg);
```

Comment:
Do not reserve memory if it can later fail. Maybe
```C
if (strlen(optarg) > MAX_OUTPUT_FILENAME) {
    ...EXIT..
}
opt.outputFile = strdup(optarg); // or strndup
```

Reply:
I can't agree more. I had it something like this initially, but changed it to the current version to follow what was already done in options.c. It seemed that most functions first assign optarg to their opt.something variable and then check it.